Makes abductor baton child of stun baton

cacogen opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
- Main change is the abductor baton's stun mode first uses the security baton's attack proc, then its own stun proc (i.e. 14 second paralyze instead of regular baton's 5 second knockdown). The only functional difference this means is that the stunned person will jitter (sprite shakes), but it removes copypasted code and allows for the abductor baton to be consistent with the regular one.
- Adds vars to stun baton for stun delay, stamina damage amount, confusion amount, off/on sound and stun sound
- Adds a different suicide if a stun baton is off or out of charge
- Adds vars to abductor baton for sleeptime, cufftime and whether the cell can be removed
- Sets abductor's baton stun force to 14 seconds, and its stun delay, cooldown, stamina damage, confusion and toggle sound to nothing (how it was before)
- Abductor baton has an infinite cell, can't be removed (how it worked before)
- Uses defines instead of magic numbers
- Makes shield check its own proc to reduce copypaste code
- Removes a lot of copypasted/duplicate code
- Renames some baton vars to make it clearer what they do or what they measure in
- Fixes a couple of typos
- Updates abductor baton typepaths in files
- Makes AbductorCheck(user) an item proc rather than an item/abductor proc to make it compatible with the refactored baton
- Makes it so you need to be an abductor to change baton modes

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game

Exposes vars for admins and balance. Reduces copypasted code. Gives the abductor baton more of the functionality of the regular baton. 
<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog
:cl:
refactor: The abductor baton is now a child of the regular stun baton. There shouldn't be a functional difference.
tweak: You need to be an abductor to change baton modes. Seemed like an oversight
spellcheck: Fixes some typos in the abductor baton's messages
admin: Both regular stun baton and abductor baton have had a lot of vars exposed pertaining to their functionality so that could come in handy I guess
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

bobbahbrown commented:
```
_maps/shuttles/emergency_zeta.dmm:70:error: undefined type: /obj/item/abductor/baton
_maps/shuttles/emergency_zeta.dmm:71:error: invalid variable name: desc
_maps/shuttles/emergency_zeta.dmm:74:error: unknown type```

map edit necessary

gitbirb commented:
i thought abductor baton was like this so that abductors are actually able to abduct people and not get memed on, and so it can combo into sleep instantly.

Jalleo commented:
Good idea. I think this is a good step. The stun baton itself may need to be setup to use components at some point.

cacogen commented:
Functionality isn't any different, download it and see for yourself.
-- 
Sent from my phone

ShizCalev commented on code/modules/antagonists/abductor/equipment/abduction_gear.dm:
```diff
@@ -430,18 +430,39 @@ Congratulations! You are now trained for invasive xenobiology research!"}
 
-/obj/item/abductor/baton/proc/toggle(mob/living/user=usr)
+	preload_cell_type = /obj/item/stock_parts/cell/infinite //Any sufficiently advanced technology is indistinguishable from magic
+	var/can_remove_cell = FALSE
```
New vars at the bottom

kriskog commented:
Merge conflict

cacogen commented on code/modules/antagonists/abductor/equipment/abduction_gear.dm:
```diff
@@ -436,22 +436,45 @@ Congratulations! You are now trained for invasive xenobiology research!"}
+
 	actions_types = list(/datum/action/item_action/toggle_mode)
 
-/obj/item/absductor/baton/ComponentInitialize()
```
>absductor

i wonder what this is causing

cacogen commented:
@kriskog
fix'd

optimumtact commented:
conflicts

AnturK commented on code/game/objects/items/stunbaton.dm:
```diff
@@ -1,34 +1,50 @@
+	throwforce = 7
+	var/throw_stun_chance = 35
+
+	var/attack_cooldown_check = (0 SECONDS)
```
What's with all the ()'s, they're not necessary.

cacogen replied:
following wrong precedent

AnturK commented on code/modules/antagonists/abductor/equipment/abduction_gear.dm:
```diff
@@ -497,45 +531,38 @@ Congratulations! You are now trained for invasive xenobiology research!"}
-							"<span class='userdanger'>[user] has stunned you with [src]!</span>")
-	playsound(src, 'sound/weapons/egloves.ogg', 50, TRUE, -1)
+/obj/item/melee/baton/abductor/attackby(obj/item/W, mob/user, params)
+	if(can_remove_cell)
```
That's too wide, extract battery removal into separate proc and check there.

AnturK reviewed (changes requested):
Looks alright just one minor change.

cacogen committed:
Makes abductor baton child of stun baton


- Clarify var names and reorder vars in baton defines
- Slightly simplify active baton wash code

cacogen committed:
forget about ()
